pkg/controller/certificaterequests/ca/ca.go

```diff
@@ -76,60 +77,47 @@ func NewCA(ctx *controllerpkg.Context) *CA {
 	}
 }
 
-func (c *CA) Sign(ctx context.Context, cr *v1alpha1.CertificateRequest) (*issuerpkg.IssueResponse, error) {
+func (c *CA) Sign(ctx context.Context, cr *v1alpha1.CertificateRequest, issuerObj v1alpha1.GenericIssuer) (*issuerpkg.IssueResponse, error) {
 	log := logf.FromContext(ctx, "sign")
+	reporter := crutil.NewReporter(log, cr, c.recorder)
 
-	issuer, err := c.helper.GetGenericIssuer(cr.Spec.IssuerRef, cr.Namespace)
-	if k8sErrors.IsNotFound(err) {
-		apiutil.SetCertificateRequestCondition(cr, v1alpha1.CertificateRequestConditionReady,
-			v1alpha1.ConditionFalse, v1alpha1.CertificateRequestReasonPending,
-			fmt.Sprintf("Referenced %s not found", apiutil.IssuerKind(cr.Spec.IssuerRef)))
+	secretName := issuerObj.GetSpec().CA.SecretName
+	resourceNamespace := c.issuerOptions.ResourceNamespace(issuerObj)
 
-		c.recorder.Event(cr, corev1.EventTypeWarning, v1alpha1.CertificateRequestReasonPending, err.Error())
-
-		log.WithValues(
-			logf.RelatedResourceNameKey, cr.Spec.IssuerRef.Name,
-			logf.RelatedResourceKindKey, cr.Spec.IssuerRef.Kind,
-		).Error(err, "failed to find referenced issuer")
-
-		return nil, nil
-	}
+	// get a copy of the CA certificate named on the Issuer
+	caCerts, caKey, err := kube.SecretTLSKeyPair(ctx, c.secretsLister, resourceNamespace, issuerObj.GetSpec().CA.SecretName)
 	if err != nil {
-		return nil, err
-	}
+		log := logf.WithRelatedResourceName(log, issuerObj.GetSpec().CA.SecretName, resourceNamespace, "Secret")
+		reporter = reporter.WithLog(log)
 
-	resourceNamespace := c.issuerOptions.ResourceNamespace(issuer)
+		if k8sErrors.IsNotFound(err) {
+			reporter.Pending(err, "MissingSecret",
+				fmt.Sprintf("Referenced secret %s/%s not found", resourceNamespace, secretName))
 
-	// get a copy of the CA certificate named on the Issuer
-	caCerts, caKey, err := kube.SecretTLSKeyPair(ctx, c.secretsLister, resourceNamespace, issuer.GetSpec().CA.SecretName)
-	if k8sErrors.IsNotFound(err) {
-		log := logf.WithRelatedResourceName(log, issuer.GetSpec().CA.SecretName, resourceNamespace, "Secret")
-		log.Info("error getting signing CA for Issuer")
+			return nil, nil
+		}
 
-		c.recorder.Event(cr, corev1.EventTypeWarning, v1alpha1.CertificateRequestReasonPending, err.Error())
+		if cmerrors.IsInvalidData(err) {
+			reporter.Pending(err, "ErrorParsingSecret",
+				fmt.Sprintf("Failed to parse key cert pair from secret %s/%s", resourceNamespace, secretName))
```

Comment:
"Failed to parse signing CA keypair from secret" instead perhaps? I don't like using the shortened form of 'certificate' in user-facing messages if it can be avoided!